### PR TITLE
Add Actions-based Fastlane Match workflow to iOS deployment guide

### PR DESCRIPTION
**Pull request by lazerwalker**
#### Changes

This expands the iOS deployment guide to include instructions for setting up Fastlane Match to run entirely on CI. This means that the whole codesigning and building for iOS process can happen on Actions, without requiring you to use a physical Mac.

Curious to hear thoughts -- this does greatly complicate this guide, but I think there's value in having a single answer to the whole process. I experimented a bunch with how to abstract this more cleanly, but couldn't come up with a better answer than "here are two separate workflows to set up match on CI and then actually run match".

#### Checklist

- [x] Read the contribution [guide](../blob/main/CONTRIBUTING.md) and accept the
      [code](../blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

**Review by davidmfinol — APPROVED** [review]
I like this additional information! I know I'll be trying to update my workflow using the info here. 

I noticed one small typo, and I think @webbertakken added some formatting rules that may need to be applied, but this otherwise looks good to me.

File: docs/03-github/06-deployment/ios.mdx [hunk 1]
```diff
@@ -8,21 +8,19 @@ complicated.
...
 
-### 1. Install Fastlane
+When you build your Unity project for iOS, Unity will produce an Xcode project that needs to be built using Xcode on a Mac. In order to upload your app to the App Store, TestFlight, or a third-party beta distribution service, you will first need to build it in Xcode and then code-sign it.
+
+We will be using a tool called [Fastlane](https://docs.fastlane.tools/getting-started/ios/setup/) to facilitate building, codesigning, and uploading your iOS build. 
 
-[Fastlane](https://docs.fastlane.tools/getting-started/ios/setup/) is a tool that can facilitate
-building, codesigning, and uploading iOS apps, and is the easiest way to deploy your Unity project
-to iOS.
+The base `fastlane` tool will manage using Xcode to build your project and directly upload those builds to Apple. The [Fastlane Match](https://docs.fastlane.tools/actions/match/) tool will generate and manage your code-signing identities and certificates for you, interacting directly with Apple via API and then storing your codesigning identities and certificates securely in a private git repo.
```
Comment (webbertakken) [inline docs/03-github/06-deployment/ios.mdx:21]:
```suggestion
`fastlane` will use Xcode to build your project and directly upload resulting builds to Apple. The [Fastlane Match](https://docs.fastlane.tools/actions/match/) tool will generate and manage your code-signing identities and certificates for you, interacting directly with Apple via API and then storing your codesigning identities and certificates securely in a private git repo.
```

File: docs/03-github/06-deployment/ios.mdx [hunk 2]
```diff
@@ -44,46 +42,147 @@ content:
...
-environments, create a new one.
+### 3. Prepare to use Fastlane Match
 
-If your Apple Developer account is messy and has lots of invalid, expired, or Xcode-managed profiles
-and certificates, you may **optionally** want to use Match to initially clean out your old developer
-portal by running `bundle exec fastlane match development` and
-`bundle exec fastlane match production`. **This will delete Apple codesigning identities and
-certificates and may break any existing workflows. It is NOT recommended if your project shares an
-Apple Developer account with other projects or teams at your company.**
+It's worth noting that the Match setup we are configuring uses a common workflow that is likely to be a good fit for many GameCI users. However, there are a number of different ways you can set up Match, and we recommend reading the Match [documentation](https://docs.fastlane.tools/actions/match/) and [codesigning guide](https://codesigning.guide) in their entirety.
```
Comment (webbertakken) [inline docs/03-github/06-deployment/ios.mdx:72]:
```suggestion
The Match setup we are configuring uses a common workflow that is likely to be a good fit for many GameCI users. However, there are a number of different ways you can set up Match, and we recommend reading the Match [documentation](https://docs.fastlane.tools/actions/match/) and [codesigning guide](https://codesigning.guide) in their entirety.
```

**Review by webbertakken — APPROVED** [review]
LGTM

Perhaps usage of ˋFastlane  vs ˋfastlaneˋ can be standardised.

I especially like how it's both descriptive and concise.

For the formatting thing please see CONTRIBUTING.md on how to set up prettier.

**Comment by lazerwalker** [comment]
Cool, I've auto-formatted with Prettier, and accepted in some of your changes. Still want to do another pass with an eye towards e.g. standardizing "fastlane" vs "Fastlane" before this is merged.

Will open a separate issue for it, but it would be really nice if the yarn CLI tasks could run the code formatting checker (and optionally provide a separate yarn command that auto-fixes) -- having to manually install and configure editor plugins seems less desirable than being able to provide built-in tooling for our autoformatters.

**Comment by davidmfinol** [comment]
In their docs, it seems like they prefer to italicize *fastlane* instead of capitalizing, so we may want to do the same.

**Comment by lazerwalker** [comment]
Cool, I’ve consistently lower-cased fastlane and fastlane match, matching their style. I suspect italicizing it doesn’t make sense unless we want to do that consistently with every similar tool name.

I think I’m happy with this if others are!

**Comment by webbertakken** [comment]
Yea agreed about italicising it. If anything we capitalise keywords, but respecting the makers brand name of course also works.

**Pull request merged**